tls/s2n_cipher_suites.c

```diff
@@ -131,6 +136,11 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                     continue;
                 }
 
+                /* Don't choose a suite the client shouldn't be able support */
+                if (conn->client_protocol_version < match->minimum_required_tls_version) {
+                    continue;
+                }
+
```

Comment:
Correction: This would help If the only cipher suites **matched by the server** for a TLSv1/TLSv1.1 client hello were defined for TLSv1.2.